### ISSUE TITLE
Bug on automatic room movement on game Sumatra Fate of Yandi

### ISSUE THREAD

**Issue by i30817** [adventuregamestudio/ags#784]
There is a animated sequence in this game that is broken in the ags3 branch. You're supposed to place the fetish statue in your inventory on the ledge on the trunk of the tree in front of you, and the kid will follow the monkey and the ego follow the kid. This works in the first room, but on the second, the script gets broken because the simulated click doesn't actually move you to the next room and you're on a 'livelock' situation with the cursor and ego movement eternally disabled.

Bug testable in this save here (created with the game engine version in wine):

[Sumatra Fate of Yandi.zip](https://github.com/adventuregamestudio/ags/files/3219102/Sumatra.Fate.of.Yandi.zip)

With this save, verified that the linux version of the same version as the wine executable distributed with the game works ( v.3.4.1.15) and ags3 branch git head fails as described. Regression test says:

```
9597fe051da70a2ef3ebdf80246eeb6fce54ce7e is the first bad commit
commit 9597fe051da70a2ef3ebdf80246eeb6fce54ce7e
Author: kmar <k_mar@seznam.cz>
Date:   Tue Apr 3 20:13:52 2018 +0200

    new jump point search routing with path refinement

:040000 040000 788aefb8397afe4d0a6e6f17ba7cea6d74ae1fec f251a09465642a1018c8b663626500244081011c M	Engine
```

**Comment by i30817** [comment]
Since this is a pathing regression i wouldn't be surprised if the new code is correct and the old code had a hack that 'made it work', but just FYI.

**Comment by i30817** [comment]
@kmar

**Comment by kmar** [comment]
Yes, there are cases where the old pathfinder would allow a 1px gap and still compute the path due to various attempts to accelerate the search. I don't know if this is the case (one would have to extract walkable areas for that particular room). If it is (as I suspect), then I wouldn't consider this a bug.

If you want backwards compatibility, you need to keep (preferably both) pathfinders and select either based on game version or with some runtime parameter. Alternatively you could use the old engine.

**Comment by sonneveld** [comment]
> I don't know if this is the case (one would have to extract walkable areas for that particular room). If it is (as I suspect), then I wouldn't consider this a bug.

Unfortunately if it's behaviour that was present before, it's not really a bug we can fix.

> If you want backwards compatibility, you need to keep (preferably both) pathfinders and select either based on game version or with some runtime parameter. Alternatively you could use the old engine.

Currently the engine is supposed to support older versions of games.  Using an older engine is a work around but shouldn't be the ultimate fix.

**Comment by i30817** [comment]
This game is not that old, so at a guess making the dev aware of the bug might be enough to get a single game fix for this particular case on a update.

**Comment by kmar** [comment]
I see. Another option would be to repair such 1px gaps at load time (like disjoint areas of the same walkable area that can be connected by adding 1px here and there), but this might possibly break other things and sounds a bit hacky.

Or simply revert to the old pathfinder. It's not my decision to make.

(from today's perspective, it'd be best to use navmeshes instead of pixel-based navigation anyway. much less space, much faster and resolution independent)

**Comment by ivan-mogilko** [comment]
We could restore old pathfinder in a separate module and run it for old games, that's an option if simplier solutions won't work with the new one.

**Comment by i30817** [comment]
The dev replied: https://itch.io/post/772111

Someone more competent at coding should probably take him on his offer to figure out how to help him avoid this bug on this particular case; that's why i posted the reply, since i'm not a AGS dev in any way.

**Comment by ivan-mogilko** [comment]
@i30817 there is a new PR which returns legacy pathfinder for older games: #904
would you be able to test it out?

**Comment by i30817** [comment]
Sure, just let me re-read how to download PRs from github without forking them again.

**Comment by ivan-mogilko** [comment]
You can register another remote at your local repository, then fetch and check out branch from there.

**Comment by i30817** [comment]
Yeah but i've found it's best to do this once so fetching works as expected in the future. I [found the trick with just a bit of googling anyway](https://gist.github.com/weppos/53d0fceba6de5cba23ff).

Then I found out i  uninstalled snap and cmake snap (the only thing i used it for) in the unrealistic expectation that ubuntu had its libssl crap handled: ` cmake : Depends: libssl1.0.0 (>= 1.0.2~beta3) but it is not installable`

which i guess was a unrealistic expectation. Small delay until it compiles now.

**Comment by i30817** [comment]
Also I have to disable the debug build trying to use the game dir to do tests with IO. Doesn't work that well without a writable dir...

**Comment by i30817** [comment]
So is this automatically selected for games started with a engine version number bellow 3.5? Because i tried the compile without doing any config on the acsetup.cfg with the buggy savegame and it worked.

**Comment by ivan-mogilko** [comment]
> Also I have to disable the debug build trying to use the game dir to do tests with IO. Doesn't work that well without a writable dir...

We must put the tests under additional compile flag.

> So is this automatically selected for games started with a engine version number bellow 3.5?

Yes it should do that.

**Comment by ivan-mogilko** [comment]
Alright, this seem to be fixed by #904 then!
